File: Right View of Binary Tree[24-July-2023]/LevelOrderTraversal/Solution.py
#  >> PYTHON CODE <<

from queue import Queue
class Solution:
    #Function to return list containing elements of right view of binary tree.
    def rightView(self,root):
        ans=[]
        q=Queue()
        
        q.put(root)

        while(not q.empty()):
            size=q.qsize()
            
            for i in range(size):
                nn=q.get()
                
                if(i==size-1): ans.append(nn.data)
                
                if(nn.left): q.put(nn.left)
                if(nn.right): q.put(nn.right)
                
        return ans

File: Right View of Binary Tree[24-July-2023]/LevelOrderTraversal/Solution.cpp
//  >> C++ CODE <<

class Solution
{
    public:
    //Function to return list containing elements of right view of binary tree.
    vector<int> rightView(Node *root)
    {
       vector<int> ans;
        if (!root)
            return ans;

        queue<Node*> q;
        q.push(root);

        while (!q.empty()) {
            int size = q.size();

            for (int i = 0; i < size; ++i) {
                Node* node = q.front();
                q.pop();

                if (i == size - 1)
                    ans.push_back(node->data);

                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
            }
        }

        return ans;

    }
};